File: lang_interface/java/com/intel/daal/services/LibraryVersionInfo.java
/* file: LibraryVersionInfo.java */
/*******************************************************************************
* Copyright 2014-2018 Intel Corporation.
*
* This software and the related documents are Intel copyrighted  materials,  and
* your use of  them is  governed by the  express license  under which  they were
* provided to you (License).  Unless the License provides otherwise, you may not
* use, modify, copy, publish, distribute,  disclose or transmit this software or
* the related documents without Intel's prior written permission.
*
* This software and the related documents  are provided as  is,  with no express
* or implied  warranties,  other  than those  that are  expressly stated  in the
* License.
*******************************************************************************/

/**
 * @defgroup services Services
 * @brief Contains classes that implement service functionality, including error handling, memory allocation, and library version information
 * @{
 */
package com.intel.daal.services;

import com.intel.daal.utils.*;
/**
 * @defgroup library_version_info Extracting Version Information
 * @brief Provides information about the version of Intel(R) Data Analytics Acceleration Library.
 * @ingroup services
 * @{
 */
/**
 *  <a name="DAAL-CLASS-SERVICES__LIBRARYVERSIONINFO"></a>
 * @brief Provides information about the version of Intel(R) Data Analytics Acceleration Library.
 * <!-- \n<a href="DAAL-REF-LIBRARYVERSIONINFO-STRUCTURE">LibraryVersionInfo structure details and Optimization Notice</a> -->
 */
public class LibraryVersionInfo {
    protected native String cGetProductStatus(long x);

    protected native String cGetBuild(long x);

    protected native String cGetBuildRev(long x);

    protected native String cGetName(long x);

    protected native String cGetProcessor(long x);

    protected native int cGetMajorVersion(long x);

    protected native int cGetMinorVersion(long x);

    protected native int cGetUpdateVersion(long x);

    protected native long cInit();

    protected long cLibraryVersionInfo;

    /** @private */
    static {
        LibUtils.loadLibrary();
    }

    /**
     *Default constructor
     */
    public LibraryVersionInfo() {
        cLibraryVersionInfo = cInit();
    }

    /**
     * Returns major library version number
     * @return Major library version
     */
    public int getMajorVersion() {
        return cGetMajorVersion(cLibraryVersionInfo);
    }

    /**
     * Returns minor library version number
     * @return Minor library version
     */
    public int getMinorVersion() {
        return cGetMinorVersion(cLibraryVersionInfo);
    }

    /**
     * Returns update library version number
     * @return Update library version
     */
    public int getUpdateVersion() {
        return cGetUpdateVersion(cLibraryVersionInfo);
    }

    /**
     * Returns product library status (alfa/beta/product)
     * @return Product library status
     */
    public String getProductStatus() {
        return cGetProductStatus(cLibraryVersionInfo);
    }

    /**
     * Returns library build
     * @return Library build
     */
    public String getBuild() {
        return cGetBuild(cLibraryVersionInfo);
    }

    /**
     * Returns library build revision
     * @return Library build revision
     */
    public String getBuildRev() {
        return cGetBuildRev(cLibraryVersionInfo);
    }

    /**
     * Returns library name
     * @return Library name
     */
    public String getName() {
        return cGetName(cLibraryVersionInfo);
    }

    /**
    * Returns the instruction set supported by the processor
     * @return the instruction set supported by the processor
     */
    public String getProcessor() {
        return cGetProcessor(cLibraryVersionInfo);
    }
}
/** @} */
/** @} */

File: examples/daal/cpp/source/services/library_version_info.cpp
/* file: library_version_info.cpp */
/*******************************************************************************
* Copyright 2015 Intel Corporation
*
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
*     http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*******************************************************************************/

/*
!  Content:
!    Intel(R) oneAPI Data Analytics Library (oneDAL) version information
!
!******************************************************************************/

/**
 * <a name="DAAL-EXAMPLE-CPP-LIBRARY_VERSION_INFO"></a>
 * \example library_version_info.cpp
 */

#include "daal.h"
#include <iostream>

using namespace daal::services;

int main() {
    LibraryVersionInfo ver;

    std::cout << "Major version:          " << ver.majorVersion << std::endl;
    std::cout << "Minor version:          " << ver.minorVersion << std::endl;
    std::cout << "Update version:         " << ver.updateVersion << std::endl;
    std::cout << "Product status:         " << ver.productStatus << std::endl;
    std::cout << "Build:                  " << ver.build << std::endl;
    std::cout << "Build revision:         " << ver.build_rev << std::endl;
    std::cout << "Name:                   " << ver.name << std::endl;
    std::cout << "Processor optimization: " << ver.processor << std::endl;

    return 0;
}
